zilliqa/src/consensus.rs

```diff
@@ -627,23 +630,16 @@ impl Consensus {
     fn apply_rewards(
         &mut self,
         committee: &[NodePublicKey],
+        parent_block: &Block,
         view: u64,
         cosigned: &BitSlice,
     ) -> Result<()> {
         debug!("apply rewards in view {view}");
 
         let rewards_per_block =
             *self.config.consensus.rewards_per_hour / self.config.consensus.blocks_per_hour as u128;
-        let block = self.head_block();
```

Comment:
It's more that the previous implementation was confusing (in the sense of "this seems obviously wrong, why is it this way") - I think my change should be logical enough. Unless Zoltan says otherwise!

Reply:
As discussed on Slack, should we remove this change and fix the issue as part of #1143 instead?